src/main/java/de/tum/in/www1/artemis/service/competency/CompetencyJOLService.java

```diff
@@ -0,0 +1,95 @@
+package de.tum.in.www1.artemis.service.competency;
+
+import static de.tum.in.www1.artemis.config.Constants.PROFILE_CORE;
+import static java.util.stream.Collectors.toMap;
+
+import java.util.Map;
+
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Service;
+
+import de.tum.in.www1.artemis.domain.competency.CompetencyJOL;
+import de.tum.in.www1.artemis.repository.CompetencyRepository;
+import de.tum.in.www1.artemis.repository.UserRepository;
+import de.tum.in.www1.artemis.repository.competency.CompetencyJOLRepository;
+import de.tum.in.www1.artemis.repository.competency.JOLValueEntry;
+import de.tum.in.www1.artemis.web.rest.errors.BadRequestAlertException;
+
+/**
+ * Service Implementation for managing CompetencyJOL.
+ */
+@Profile(PROFILE_CORE)
+@Service
+public class CompetencyJOLService {
+
+    private final String ENTITY_NAME = "CompetencyJOL";
+
+    private final CompetencyJOLRepository competencyJOLRepository;
+
+    private final CompetencyRepository competencyRepository;
+
+    private final UserRepository userRepository;
+
+    public CompetencyJOLService(CompetencyJOLRepository competencyJOLRepository, CompetencyRepository competencyRepository, UserRepository userRepository) {
+        this.competencyJOLRepository = competencyJOLRepository;
+        this.competencyRepository = competencyRepository;
+        this.userRepository = userRepository;
+    }
+
+    private static boolean isJolValueValid(int jolValue) {
```

Comment:
Can we get a java doc here why this is the valid range?

Reply:
Additionally we could move these magic numbers to constants.